src/App.tsx

```diff
@@ -2,15 +1,63 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Dropdown } from './components/Dropdown';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface Props {
+  debounceDelay?: number;
+}
+
+export const App: React.FC<Props> = ({ debounceDelay = 300 }) => {
+  const [query, setQuery] = useState('');
+  const [focused, setFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyFilter = useMemo(
+    () =>
+      debounce((currentQuery: string) => {
+        setQuery(currentQuery);
+      }, debounceDelay),
+    [debounceDelay],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(query.toLowerCase()),
```

Comment:
The filtering logic does not handle cases where the query consists of spaces only. Consider trimming the query or checking for non-space characters before applying the filter.